pkgs/by-name/x2/x2t/package.nix

```diff
@@ -0,0 +1,344 @@
+{ 
+  stdenv,
+  fetchFromGitHub,
+  pkg-config,
+  libxml2,
+  boost,
+  gumbo,
+  icu,
+  freetype,
+  qt5,
+  harfbuzz,
+  # needs to be static and built with MD2 support!
+  openssl,
+  # should find out if newer versions work...
+  v8,
+  # cheating a bit ;)
+  onlyoffice-documentserver
+}:
+
+let
+  # see core/Common/3dParty/fetch.sh
+  katana-parser-src = fetchFromGitHub {
+    owner = "jasenhuang";
+    repo = "katana-parser";
+    rev = "be6df458d4540eee375c513958dcb862a391cdd1";
+    hash = "sha256-SYJFLtrg8raGyr3zQIEzZDjHDmMmt+K0po3viipZW5c=";
+  };
+  # see build_tools scripts/core_common/modules/hyphen.py
+  hyphen-src = fetchFromGitHub {
+    owner = "hunspell";
+    repo = "hyphen";
+    rev = "73dd2967c8e1e4f6d7334ee9e539a323d6e66cbd";
+    hash = "sha256-WIHpSkOwHkhMvEKxOlgf6gsPs9T3xkzguD8ONXARf1U=";
+  };
+  # see core/Common/3dParty/html/fetch.py
+  gumbo-parser-src = fetchFromGitHub {
+    owner = "google";
+    repo = "gumbo-parser";
+    rev = "aa91b27b02c0c80c482e24348a457ed7c3c088e0";
+    hash = "sha256-+607iXJxeWKoCwb490pp3mqRZ1fWzxec0tJOEFeHoCs=";
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "x2t";
+  version = "8.1.1";
+
+  src = fetchFromGitHub {
+    owner = "ONLYOFFICE";
+    repo = "core";
+    # corresponds to documentserver 8.1.1
+    rev = "90d46b55d8d10814f961661f815189e16d2271b9";
+    hash = "sha256-1gH3AFYE4v21+3C4QKboHKaP0XmEN7gr1lG8+pT4SC4=";
+  };
+  patchFlags = [ "-p1" "-l" ];
+  patches = [
+    # https://github.com/ONLYOFFICE/core/pull/1631
+    ./doctrenderer-format-security.patch
+    ./fontengine-format-security.patch
+    # Not sure what's going on here...
+    ./MsBinaryFile-pragma-regions.patch
+    # Interestingly only seems to pop up when debug mode is enabled
+    ./xlsx-missing-import.patch
+  ];
+
+  nativeBuildInputs = [ pkg-config qt5.full ];
+  buildInputs = [ gumbo boost libxml2.dev icu freetype.dev ];
+  dontStrip = true;
+  buildPhase = ''
+    runHook preBuild
+
+    # https://github.com/onlyOFFICE/build_tools makes many assumptions,
+    # so we do things 'manually' here...
+
+    substituteInPlace \
+      Common/3dParty/icu/icu.pri \
+      --replace-fail "ICU_MAJOR_VER = 58" "ICU_MAJOR_VER = 74"
+
+    # https://github.com/ONLYOFFICE/core/pull/1637
+    # (but not as patch because line endings)
+    substituteInPlace \
+      UnicodeConverter/UnicodeConverter.cpp \
+      --replace-fail "TRUE" "true"
+
+    substituteInPlace \
+      DesktopEditor/doctrenderer/nativecontrol.h \
+      --replace-fail "fprintf(f, strVal.c_str());" "fprintf(f, \"%s\", strVal.c_str());" \
+      --replace-fail "fprintf(_file, sParam.c_str());" "fprintf(_file, \"%s\", sParam.c_str());"
+
+    substituteInPlace \
+      ./Common/Network/FileTransporter/src/manager.cpp \
+      --replace-fail "../core" ""
+    substituteInPlace \
+      ./Test/Applications/TestDownloader/mainwindow.h \
+      --replace-fail "../core" ""
```

Comment:
(these are not patches because of issues with line endings, even with patch -l - should probably make it upstream anyway)